pkg/kv/kvserver/mvcc_gc_queue.go

```diff
@@ -425,13 +450,26 @@ func makeMVCCGCQueueScoreImpl(
 
 	// If range was deleted, lower score threshold to allow faster cleanup as
 	// most of the garbage is concentrated at the ttl threshold.
-	if !r.ShouldQueue && suspectedFullRangeDeletion(ms) {
-		r.ShouldQueue = canAdvanceGCThreshold && r.FuzzFactor*valScore > mvccGCDropRangeKeyScoreThreshold
+	if suspectedFullRangeDeletion(ms) {
+		if !r.ShouldQueue {
+			r.ShouldQueue = canAdvanceGCThreshold && r.FuzzFactor*valScore > mvccGCDropRangeKeyScoreThreshold
+		}
+		// If we are removing all data from the range, then we should check if we
+		// have a GC hint that would trigger scanning for related ranges and raise
+		// priority to collect all related ranges together.
+		if r.ShouldQueue && gcHintedRangeDelete(hint, gcTTL, now) {
```

Comment:
I'm slightly worried that we can end up with a range that has a hint that can't be clear range'd because of some bug in split/merge logic and we'll end up enqueueing them with high priority.
We have a mitigation to throttle that though.
So maybe use hint alone if we have estimates and use stats if they are accurate and tell us we have candidate? I think that would be inbetween and gives us safer approach.